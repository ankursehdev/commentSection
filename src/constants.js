export const commentDetails = {
  authorName: "Brad",
  author: "Author",
  commentText:
    "Lorem ipsum dolor sit amet, consectetur adipisicing elit. Id veniam quia distinctio temporibus delectus laborum, error qui iusto recusandae nesciunt voluptatem cupiditate.",
  time: "2 Minutes ago",
  numberOfReplies: 21,
  upVotes: 123,
  downVotes: 0,
};
