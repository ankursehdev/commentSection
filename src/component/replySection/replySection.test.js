import React from "react";
import { render, cleanup } from "@testing-library/react";
import ReplyCount from "./index";

import { commentDetails as response } from "../../constants";

afterEach(cleanup);

test("Render Reply section properly", () => {
  render(<ReplyCount type="count" count={response.numberOfReplies} />);
});

test("Render Reply section with down type", () => {
  const { getByText } = render(
    <ReplyCount type="down" count={response.downVotes} />
  );
  const downVotesText = getByText(`${response.downVotes}`);
  expect(downVotesText).toBeInTheDocument();
});

test("Render Reply section with up type", () => {
  const { getByText } = render(
    <ReplyCount type="up" count={response.upVotes} />
  );
  const upVotesText = getByText(`${response.upVotes}`);
  expect(upVotesText).toBeInTheDocument();
});

test("Taking Count snapshot", () => {
  const { container } = render(
    <ReplyCount type="count" count={response.numberOfReplies} />
  );
  expect(container).toMatchSnapshot();
});
