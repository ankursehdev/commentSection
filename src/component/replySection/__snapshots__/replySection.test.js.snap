// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Taking Count snapshot 1`] = `
<div>
  <button>
    <span
      class="reply-count"
    >
      21
    </span>
     Replies
  </button>
</div>
`;
